crates/astria-sequencer-relayer/tests/blackbox/helper.rs

```diff
@@ -78,6 +99,31 @@ const PRIVATE_VALIDATOR_KEY: &str = r#"
 }
 "#;
 
+pub struct MockSequencerServer {
+    blocks: Arc<Mutex<VecDeque<SequencerBlock>>>,
```

Comment:
Could this just be a `tokio::sync::mpsc::UnboundedReceiver` instead of this relatively complex `Arc<Mutex<VecDeque<>>>`?

Reply:
tried that initially but it requires `&mut self` in the grpc methods, which doesn't work as the trait requires `&self`